solr/test-framework/src/java/org/apache/solr/SolrJettyTestBase.java

```diff
@@ -138,17 +142,15 @@ public static void afterSolrJettyTestBase() throws Exception {
     solrClientTestRule.reset();
   }
 
-  /**
-   * Create a new solr client. If createJetty was called, a http implementation will be created,
-   * otherwise an embedded implementation will be created. Subclasses should override for other
-   * options.
-   */
-  public SolrClient createNewSolrClient() {
-    return solrClientTestRule.getSolrClient();
+  public synchronized SolrClient getSolrClient() {
+    if (getClient() == null) {
+      return solrClientTestRule.getSolrClient();
+    }
+    return getClient();
   }
```

Comment:
When the method fundamentally changes away from using fields on this class to calling the rule, it no longer needs to be synchronized because the rule's getSolrClient is thread-safe.

also, don't see why getClient exists nor why it would return null.